osmnx/io.py

```diff
@@ -419,8 +419,25 @@ def _convert_node_attr_types(G, dtypes=None):
     Returns
     -------
     G : networkx.MultiDiGraph
+    
+    Modified to include detection of list, dict, and set type
+    node attribute strings and handle their translation into
+    python objects of the correct type.
     """
     for _, data in G.nodes(data=True):
+        # first, eval stringified lists, dicts or sets to convert them to objects
+        # edge attributes might have a single value, or a list if simplified
+        for attr, value in data.items():
+            # check for stringified lists
+            if value.startswith("[") and value.endswith("]"):
+                with contextlib.suppress(SyntaxError, ValueError):
+                    data[attr] = ast.literal_eval(value)
+            # check for stringified dicts or sets: both can be converted
+            # to python objects using ast.literal_eval()
+            if value.startswith("{") and value.endswith("}"):
+                with contextlib.suppress(SyntaxError, ValueError):
+                    data[attr] = ast.literal_eval(value)
+    
```

Comment:
We previously only ran `literal_eval` on edge attributes. Is there a use case for node attributes? OSMnx does not produce any node attributes of type list, dict, or set.

Reply:
I can describe my usage, if it makes sense to you.  It's clearly all custom attributes, as I mentioned in the issue description, so not necessarily OSMnx scope.

I'm doing 15-minute city metrics, based on OSMnx with a lot of my own code on top.  I fetch a walking network via OSMnx, then integrate amenity POI nodes into it, adding linking edges with lengths.  The network nodes, apart from the amenities, are considered to be "residential locations", without getting into filtering details.  Processing then obtains, among other things, shortest path distances, per residential node, to each amenity node.  These are stored per node into a dict of dicts attribute keyed by amenity category and amenity node.  
Accessible amenities are counted per residential node and edge, also, which allows for producing isochrone buffered geometries, for example.

It's true that I could have managed these outside the graph object using dicts, but I had wanted to make the work graph-oriented, and I believe it simplifies things, avoiding the need to keep separate objects consistent when doing things like truncating, etc, etc.